mafiasi/gprot/views.py

```diff
@@ -455,3 +458,27 @@ def delete_reminder(request, reminder_pk):
     reminder.delete()
 
     return redirect('gprot_reminders')
+
+@login_required
+def favorite(request):
+    action = request.POST.get('action', None)
+    user = request.user
+    url = request.POST.get('url', None)
+    if action == 'save':
+        if Favorite.objects.filter(user=user, url=url):
+              return HttpResponse('already a favorite', status=400)
+        favorite = Favorite.objects.create(user=user, url=url)
+        keywords = re.split("search=", url)[1:]
+        for keyword in keywords:
+            if keyword[-1] == "&":
+                keyword = keyword[:-1]
+            if keyword[0] == "c":
+                favorite.courses.add(Course.objects.get(id=int(keyword[2:])))
+            elif keyword[0] == "e":
+                favorite.examiners.add(Teacher.objects.get(id=int(keyword[2:])))
+    elif action == 'delete':
+        favorite = get_object_or_404(Favorite, user=user, url=url)
+        favorite.delete()
+    else:
+        return HttpResponse(status=400)
+    return HttpResponse(status=200)
```

Comment:
The clean thing here (and typical for REST apis) would be to return `201` when it is created and `204` when it was deleted.